src/mpid/ch4/shm/posix/eager/fbox/fbox_recv.h

```diff
@@ -0,0 +1,149 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2017 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+#ifndef POSIX_EAGER_FBOX_RECV_H_INCLUDED
+#define POSIX_EAGER_FBOX_RECV_H_INCLUDED
+
+#include "fbox_impl.h"
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_POSIX_eager_recv_begin
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int
+MPIDI_POSIX_eager_recv_begin(MPIDI_POSIX_eager_recv_transaction_t * transaction)
+{
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_POSIX_FBOX_EAGER_RECV_BEGIN);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_POSIX_FBOX_EAGER_RECV_BEGIN);
+
+    enum { BATCH_SIZE = 4 };
+    int j, local_rank, grank;
+    MPIDI_POSIX_fastbox_t *fbox_in;
+    int mpi_errno = MPIDI_POSIX_NOK;
+
```

Comment:
You mean we should set the values to `NULL` (or whatever makes sense) when we don't find something? I'm not sure that's necessary since we indicate that with the return value.

Reply:
Oh, I missed the return value part. Then it is OK.